From 773625547c9d488d67e2c5721e9b5c0e81ba85be Mon Sep 17 00:00:00 2001
From: Rohan Garg <rohan@garg.io>
Date: Thu, 13 Dec 2018 17:46:27 +0100
Subject: [PATCH 12/29] glamor/egl: Drop a implicit depends on
 EGL_KHR_no_config_context

Not all platforms provide EGL_KHR_no_config_context, this ensures
that eglCreateContext works on such platforms by querying for
a config.

(cherry picked from commit 88dd4cea51f7e461cb4859c06bfe641890caa97b)
---
 glamor/glamor_egl.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/glamor/glamor_egl.c b/glamor/glamor_egl.c
index 51c5437a7..fa854d65f 100644
--- a/glamor/glamor_egl.c
+++ b/glamor/glamor_egl.c
@@ -905,6 +905,8 @@ glamor_egl_init(ScrnInfoPtr scrn, int fd)
 {
     struct glamor_egl_screen_private *glamor_egl;
     const GLubyte *renderer;
+    EGLConfig egl_config;
+    int n;
 
     glamor_egl = calloc(sizeof(*glamor_egl), 1);
     if (glamor_egl == NULL)
@@ -983,8 +985,14 @@ glamor_egl_init(ScrnInfoPtr scrn, int fd)
             goto error;
         }
 
+        if (!eglChooseConfig(glamor_egl->display, NULL, &egl_config, 1, &n)) {
+            xf86DrvMsg(scrn->scrnIndex, X_ERROR,
+                       "glamor: No acceptable EGL configs found\n");
+            goto error;
+        }
+
         glamor_egl->context = eglCreateContext(glamor_egl->display,
-                                               NULL, EGL_NO_CONTEXT,
+                                               egl_config, EGL_NO_CONTEXT,
                                                config_attribs);
     }
     if (glamor_egl->context == EGL_NO_CONTEXT) {
-- 
2.11.0

